File: backend/src/main/java/com/back/ourlog/domain/report/dto/ReportRequest.java
package com.back.ourlog.domain.report.dto;

import com.back.ourlog.domain.report.entity.ReportReason;
import jakarta.persistence.Column;
import jakarta.validation.constraints.NotNull;

public record ReportRequest(
        @NotNull Integer targetUserId,
        @NotNull ReportReason type,
        @NotNull @Column(length = 255) String description
) {}

File: backend/src/main/java/com/back/ourlog/domain/report/dto/ReportRequest.kt
package com.back.ourlog.domain.report.dto

import com.back.ourlog.domain.report.entity.ReportReason
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size

data class ReportRequest(
    @field:NotNull
    val targetUserId: Int,

    @field:NotNull
    val type: ReportReason,

    @field:NotNull
    @field:Size(max = 255)
    val description: String
)
